model/common/src/icon4py/model/common/decomposition/mpi_decomposition.py

```diff
@@ -204,6 +223,174 @@ def exchange_and_wait(self, dim: Dimension, *fields: tuple):
+            sdfg = dace.SDFG("_halo_exchange_")
+            state = sdfg.add_state()
+
+            global_buffers = {
+                self.__sdfg_signature__()[0][i]: arg for i, arg in enumerate(args)
+            }  # Field name : Data Descriptor
+
+            halo_exchange.add_halo_tasklet(
+                sdfg, state, global_buffers, self, dim, id(self), wait, self.num_of_halo_tasklets
+            )
+
+            sdfg.arg_names.extend(self.__sdfg_signature__()[0])
+            sdfg.arg_names.extend(list(self.__sdfg_closure__().keys()))
+
+            self.num_of_halo_tasklets += 1
+            return sdfg
+
+        def dace__sdfg_closure__(
+            self, reevaluate: Optional[dict[str, str]] = None
+        ) -> dict[str, Any]:
+            # Get the underlying C++ pointers of the GHEX objects and use them in the halo exchange tasklet
+            return {ghex_ptr_name: dace.uintp for ghex_ptr_name in halo_exchange.ghex_ptr_names}
+
+        def dace__sdfg_signature__(self) -> tuple[Sequence[str], Sequence[str]]:
+            args = []
+            for i in range(GHexMultiNodeExchange.max_num_of_fields_to_communicate_dace):
+                args.append(f"field_{i}")
+            return (args, [])
+    else:
+
+        def dace__sdfg__(self, *args, **kwargs):
+            raise NotImplementedError(
+                "__sdfg__ is only supported when the 'dace' module is available."
+            )
+
+        def dace__sdfg_closure__(
+            self, reevaluate: Optional[dict[str, str]] = None
+        ) -> dict[str, Any]:
+            raise NotImplementedError(
+                "__sdfg_closure__ is only supported when the 'dace' module is available."
+            )
+
+        def dace__sdfg_signature__(self) -> tuple[Sequence[str], Sequence[str]]:
+            raise NotImplementedError(
+                "__sdfg_signature__ is only supported when the 'dace' module is available."
+            )
+
+    __sdfg__ = dace__sdfg__
+    __sdfg_closure__ = dace__sdfg_closure__
+    __sdfg_signature__ = dace__sdfg_signature__
+
+
+@dataclass
+class HaloExchangeWait:
+    exchange_object: GHexMultiNodeExchange
+
+    buffer_name: ClassVar[str] = "communication_handle"  # DaCe-related
+
+    def __call__(self, communication_handle: MultiNodeResult):
+        communication_handle.wait()
+
+    if dace:
+        # Implementation of DaCe SDFGConvertible interface
+        def dace__sdfg__(self, *args, **kwargs):
+            sdfg = dace.SDFG("_halo_exchange_wait_")
+            state = sdfg.add_state()
+
+            # The communication handle used in the halo_exchange tasklet is a global variable
+            # ghex::communication_handle<communication_handle_type> h_{id(self.exchange_object)};
+            # Therefore, this tasklet calls the wait() method on the communication handle -disregards any input-
+            tasklet = dace.sdfg.nodes.Tasklet(
+                "_halo_exchange_wait_",
+                inputs=None,
```

Comment:
Same comment applies to multiple places: `inputs` and `outputs` should be set of strings representing the names of input and output connectors, respectively:
```suggestion
                inputs={f"IN_{buffer_name}"},
                outputs={"__out"},
```

Reply:
you are right that I could write it like this, but I prefer to do a simple initialisation there and afterwards I set the inputs/outputs. I just maintain the same approach as the `GHexMultiNodeExchange`.